libdrgn/orc_info.c

```diff
@@ -355,6 +356,164 @@ static struct drgn_error *drgn_read_orc_sections(struct drgn_module *module)
+			 uint64_t unwind, uint64_t unwind_ip, uint64_t header)
+{
+	uint8_t header_data[20];
+
+	struct drgn_error *err;
+
+	if (header) {
+		err = drgn_program_read_memory(module->prog, header_data,
+					       header, sizeof(header_data),
+					       false);
+
+		if (err)
+			return err;
+
+		module->orc.version = orc_version_from_header(header_data);
+		if (module->orc.version < 0)
+			return drgn_error_create(DRGN_ERROR_OTHER,
+							"unrecognized .orc_header");
+	} else {
+		module->orc.version = orc_version_from_osrelease(module->prog);
+	}
+
+	_cleanup_free_ int32_t *pc_offsets = malloc_array(num_entries,
+							  sizeof(pc_offsets[0]));
+	err = drgn_program_read_memory(module->prog, pc_offsets, unwind_ip,
+				       num_entries * sizeof(pc_offsets[0]), false);
+	if (err)
+		return err;
+
+	_cleanup_free_ struct drgn_orc_entry *entries =
+		malloc_array(num_entries, sizeof(entries[0]));
+	err = drgn_program_read_memory(module->prog, entries, unwind,
+				       num_entries * sizeof(entries[0]), false);
+	if (err)
+		return err;
+
+	module->orc.entries = no_cleanup_ptr(entries);
+	module->orc.pc_offsets = no_cleanup_ptr(pc_offsets);
+	module->orc.num_entries = num_entries;
+	module->orc.pc_base = unwind_ip;
+	drgn_log_debug(module->prog, "Loaded built-in ORC (v%d) for module %s",
+		       module->orc.version, module->name);
+	return NULL;
+}
+
+static struct drgn_error *drgn_read_vmlinux_orc(struct drgn_module *module)
+{
+	struct drgn_error *err;
+	struct drgn_symbol *sym;
+
+	uint64_t unwind_ip_start, unwind_ip_end;
+	uint64_t unwind_start, unwind_end;
+	uint64_t header_start = 0, header_end = 0;
+
+#define get_symbol(name, var, optional) \
+        err = drgn_program_find_symbol_by_name(module->prog, name, &sym); \
+        if (!err) { \
+                var = sym->address; \
+                drgn_symbol_destroy(sym); \
+                sym = NULL; \
+        } else if (optional && err->code == DRGN_ERROR_LOOKUP) { \
+                drgn_error_destroy(err); \
+                sym = NULL; \
+                err = NULL; \
+        } else { \
+                return err; \
+        }
+
+        get_symbol("__start_orc_unwind_ip", unwind_ip_start, false);
+        get_symbol("__stop_orc_unwind_ip", unwind_ip_end, false);
+        get_symbol("__start_orc_unwind", unwind_start, false);
+        get_symbol("__stop_orc_unwind", unwind_end, false);
+        get_symbol("__start_orc_header", header_start, true);
+        get_symbol("__stop_orc_header", header_end, true);
+#undef get_symbol
+
+	if ((unwind_ip_end - unwind_ip_start) % sizeof(int32_t))
+		return drgn_error_create(DRGN_ERROR_OTHER, "invalid built-in orc_unwind_ip range");
+	uint64_t num_entries = (unwind_ip_end - unwind_ip_start) / sizeof(int32_t);
+	if (num_entries > UINT_MAX)
+		return drgn_error_create(DRGN_ERROR_OTHER,
+					 "built-in orc_unwind_ip range is too large");
+
+	if ((unwind_end - unwind_start) % sizeof(struct drgn_orc_entry)	\
```

Comment:
No need for  this backslash.
```suggestion
	if ((unwind_end - unwind_start) % sizeof(struct drgn_orc_entry)
```